### PR TITLE
Enable Mypy Check in torch/_inductor/select_algorithm.py

### PR DESCRIPTION
**Pull request by fffrog**
Fixes #105230 to enable mypy check.

```
$ mypy  torch/_inductor/select_algorithm.py
Success: no issues found in 1 source file
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

**Comment by zou3519** [comment]
mypy seems to be failing

File: torch/_inductor/select_algorithm.py [hunk 1]
```diff
@@ -887,7 +887,7 @@ def tensor_repr(x):
             lines += ["]", f"out = {tensor_repr(out)}", ""]
             return "\n".join(lines)
 
-        benchmark.debug_str = debug_str
+        setattr(benchmark, "debug_str", debug_str) # noqa: B010
```
Comment (Skylion007) [inline torch/_inductor/select_algorithm.py:890]: Why add this bug prone behavior? Just add a type ignore or a typing.cast if you really cannot fix it.

**Review by Skylion007 — COMMENTED** [review]
Mypy is failing as well

File: torch/_inductor/select_algorithm.py (hunk 1 above)
Comment (FFFrog) [reply torch/_inductor/select_algorithm.py]: Ok, just added a type ignore

**Comment by FFFrog** [comment]
> mypy seems to be failing

Sorry, I made a mistake and will fix again

**Review by eellison — APPROVED** [review]
nice !

**Pull request closed without merging**